app/assets/scripts/components/project-timeline.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+import React from 'react';
+import { scaleTime } from 'd3';
+import { parseProjectDate, formatDate } from '../utils/date';
+
+var ProjectTimeline = React.createClass({
+  displayName: 'ProjectTimeline',
+
+  propTypes: {
+    project: React.PropTypes.object
+  },
+
+  render: function () {
+    const project = this.props.project;
+
+    const plannedStart = parseProjectDate(project.planned_start_date);
+    const plannedEnd = parseProjectDate(project.planned_end_date);
+
+    const actualStart = project.actual_start_date ? parseProjectDate(project.actual_start_date) : null;
+    const actualEnd = project.actual_end_date ? parseProjectDate(project.actual_end_date) : null;
+
+    const start = Math.min(plannedStart, actualStart);
+    const end = Math.max(plannedEnd, actualEnd);
+
+    const scale = scaleTime().domain([new Date(start), new Date(end)]).range([0, 100]);
+
+    return (
+      <div className='timeline'>
+        <div className='timeline__unit'>
+          <h5>Current Progress</h5>
+          {actualStart && actualEnd && timeline(actualStart, actualEnd, scale)}
```

Comment:
@dereklieu when this doesn't exist, should we show "No data"? Right now there's nothing displayed.